src/services/dimension-processor.ts

```diff
@@ -310,14 +229,58 @@ async function recreateBaseFactTable(dataset: Dataset, dataTable: DataTable): Pr
     }
 }
 
+async function removeAllDimensions(dataset: Dataset) {
+    logger.warn(`Removing all dimensions for dataset ${dataset.id}`);
+    for (const dimension of dataset.dimensions) {
+        if (dimension.lookupTable) {
+            const dataLakeService = new DataLakeService();
+            try {
+                dataLakeService.deleteFile(dimension.lookupTable.filename, dataset.id);
+            } catch (error) {
+                logger.warn(
+                    error,
+                    `Something went wrong trying to remove previously uploaded lookup table with error: ${error}`
+                );
+            }
+            logger.warn(`Removing lookup table for dimension ${dimension.id}`);
+            await LookupTable.getRepository().delete({ dimension });
+        }
+        await DimensionMetadata.getRepository().delete({ dimension });
```

Comment:
We've got an `onDelete: 'CASCADE'` on the metadata side of this join - shouldn't they get automatically be deleted when you remove the dimension record?

Maybe it only happens with `.remove()` and not `.delete()`? 

We don't have an `onDelete` on the LookupTable - is it worth adding?

Reply:
Yes we probably should although I wonder how it will work as both measure and dimension both use the same lookup table entity.

Reply:
you can have the same lookup table row referenced by both a dimension and a measure?